src/modules/aws_s3/entity/public_access_block.ts

```diff
@@ -0,0 +1,83 @@
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+
+import { cloudId } from '../../../services/cloud-id';
+import { Bucket } from './bucket';
+
+/**
+ * Table to manage AWS S3 bucket public access.
+ *
+ * A bucket is a container for objects stored in Amazon S3. You can store any number of objects in a bucket and can have up to 100 buckets in your account.
+ *
+ * @example
+ * ```sql TheButton[Make the S3 Bucket Public]="Make the S3 Bucket Public"
+ * UPDATE public_access_block SET BlockPublicAcls = false, IgnorePublicAcls = false, BlockPublicPolicy = false, RestrictPublicBuckets = false WHERE bucket_name = 'mybucketname';
+ * ```
+ *
+ * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-control-block-public-access.html
+ */
+@Entity()
+export class PublicAccessBlock {
```

Comment:
can we add the auto-incremented id?

Reply:
Here there's a 1-1 relationship between `Bucket` and `PublicAccessBlock` so I believe `bucket` field is enough (and has a unique constraint - it's primary key).